finagle-http/src/test/scala/com/twitter/finagle/http/RequestTest.scala

```diff
@@ -65,6 +65,18 @@ class RequestTest extends FunSuite {
     assert(actual == expected)
   }
 
+  test("Default Host header for HTTP_1.1") {
```

Comment:
Could you add a test that verifies that the Host header isn't overwritten if its already present?